### DISCUSSION TITLE
Data Resource path that work both as a relative path on disk and on online

### DISCUSSION THREAD

**Discussion by rgaiacs** [frictionlessdata/frictionlessdata.io#747] [Q&A]
One of the examples at https://specs.frictionlessdata.io/data-resource/ is

```
# relative path
# note: this will work both as a relative path on disk and on online
"path": "my-data-directory/my-csv.csv"
```

Is this related to to release < 1.0.0-beta.18 (Nov 17 2016) that have a `url` property?

**Comment by roll** [comment]
I think it means that whereever the datapackage is it will work the same e.g. if it's on github the basepath will be a remote dir but the file path will be resolved correctly anyway

**Comment by rgaiacs** [comment]
@roll Thanks for the reply. I searched how to define the `basepath` but I didn't find it. Is the `basepath` leave for the client to define? After study the [reference R implementation](https://github.com/frictionlessdata/frictionless-r), the client implements the `basepath` (see `read_from` in https://github.com/frictionlessdata/frictionless-r/blob/c0869654c1bae1d25782c0315fa81f96fa9fa291/R/get_resource.R#L45).

**Reply by roll** [reply]
Yes `basepath` has to be handled by Software. The location of descriptor define a basepath e.g.:

descriptor: https://raw.githubusercontent.com/datasets/oil-prices/master/datapackage.json
basepath: https://raw.githubusercontent.com/datasets/oil-prices/master/
fullpath of a resource: basepath + "data/brent-daily.csv"
